bundles/org.openhab.ui/web/src/components/widgets/system/oh-knob.vue

```diff
@@ -1,33 +1,35 @@
 <template>
-  <div>
-    <knob-control v-if="!config.useSliderControl" v-bind="config" :text-color="config.textColor || (($f7.data.themeOptions.dark === 'dark') ? '#ffffff' : undefined)" :value="value"
-                  @input="sendCommandDebounced($event)" @click.native.stop="sendCommandDebounced(value, true)" @touchend.native.stop="sendCommandDebounced(value, true)" />
-
-    <round-slider v-else :value="value" :min="config.min" :max="config.max" :step="config.stepSize" :radius="config.size/2"
-                  :style="`stroke-dasharray: ${(config.dottedPath) ? config.dottedPath : 0}`"
-                  :rangeColor="config.primaryColor" :pathColor="config.secondaryColor"
-                  :tooltipColor="config.textColor || (($f7.data.themeOptions.dark === 'dark') ? '#ffffff' : undefined)"
-                  :disabled="config.disabled" :width="config.strokeWidth" :line-cap="config.lineCap"
-                  :startAngle="config.startAngle" :endAngle="config.endAngle"
-                  :borderWidth="config.borderWidth" :borderColor="config.borderColor" :circleShape="config.circleShape"
-                  @input="sendCommandDebounced($event)" @click.native.stop="sendCommandDebounced(value, true)" @touchend.native.stop="sendCommandDebounced(value, true)" />
-  </div>
+  <round-slider v-bind="resolvedConfig" :value="value" :style="`stroke-dasharray: ${(config.dottedPath) ? config.dottedPath : 0}`" mouseScrollAction="true"
+                @input="sendCommandDebounced($event)" @click.native.stop="sendCommandDebounced(value, true)" @touchend.native.stop="sendCommandDebounced(value, true)" />
 </template>
 
 <script>
 import mixin from '../widget-mixin'
 import slideMixin from './slide-mixin'
 import { OhKnobDefinition } from '@/assets/definitions/widgets/system'
 
-import KnobControl from 'vue-knob-control'
-import RoundSlider from 'vue-round-slider'
-
 export default {
   mixins: [mixin, slideMixin],
   components: {
-    KnobControl,
-    RoundSlider
+    // See https://roundsliderui.com/document.html for docs
+    'RoundSlider': () => import(/* webpackChunkName: "vue-round-slider" */ 'vue-round-slider')
   },
-  widget: OhKnobDefinition
+  widget: OhKnobDefinition,
+  computed: {
+    resolvedConfig () {
+      const cfg = this.config
+      return {
+        ...cfg,
+        step: cfg.step ? cfg.step : cfg.stepSize,
+        radius: cfg.radius ? cfg.radius : (cfg.responsive ? cfg.size / 2 + '%' : cfg.size / 2),
+        rangeColor: cfg.rangeColor ? cfg.rangeColor : cfg.primaryColor,
+        pathColor: cfg.pathColor ? cfg.pathColor : cfg.secondaryColor,
+        tooltipColor: cfg.tooltipColor ? cfg.tooltipColor : cfg.textColor,
+        width: cfg.width ? cfg.width : cfg.strokeWidth,
+        startAngle: !cfg.circleShape ? (cfg.startAngle ? cfg.startAngle : -50) : null,
+        endAngle: !cfg.circleShape ? (cfg.endAngle ? cfg.endAngle : -130) : null
```

Comment:
Please be aware of the pitfalls of these ternary syntaxes for fallback values, any falsy value doesn't necessarily have to fallback. It's not "if undefined then.... else", it's "if falsy then... else".

For instance if your `cfg.radius` is 0 for some reason then the fallback would take effect, even if 0 would have been a valid and reasonable value.
Perhaps you're actually aware and I apologize for being pedantic, but I did bite me a few times so just wanted to make sure.

Reply:
I am aware of false and it should not end up in these paramaters, but I haven't thought of `0`. I will fix the fallbacks, thanks.

Reply:
I've fixed the fallbacks and now explicitely check for `undefined` see https://github.com/openhab/openhab-webui/pull/1718/commits/6b6523e721c926a53973b6be152ab971ab6ac11d.
Note that for `responsive` the ternary operator is fine using falsy.